frontend/pages/beta/index.tsx

```diff
@@ -0,0 +1,164 @@
+  }
+
+  const updateSelectedType = (selectedType: string) => {
+    if (selectedType) {
+      switch (selectedType) {
+        case 'My models':
+          setGroup('user')
+          break
+        case 'Favourites':
+          setGroup('favourites')
+          break
+        default:
+          setGroup('all')
+      }
+      mutateModels()
+    }
+  }
+
+  return (
+    <Wrapper title='Explore Models' page='marketplace'>
+      <Paper
+        component='form'
+        onSubmit={onFilterSubmit}
+        sx={{
+          p: '2px 4px',
+          display: 'flex',
+          alignItems: 'center',
+          width: '70%',
+          maxWidth: '400px',
+          marginBottom: 3,
+        }}
+      >
+        <InputBase sx={{ ml: 1, flex: 1 }} placeholder='Filter Models' value={filter} onChange={handleFilterChange} />
+        <IconButton color='primary' type='submit' sx={{ p: '10px' }} aria-label='filter'>
+          <SearchIcon />
+        </IconButton>
+      </Paper>
+      <Grid container>
+        <Grid item sm={8} xs={12}>
+          <Paper sx={{ py: 2, px: 4 }}>
+            <Box sx={{ mb: 2, borderBottom: 1, borderColor: 'divider' }} data-test='indexPageTabs'>
+              <Tabs value={'bailo'}>
+                <Tab label={`Models ${models ? `(${models.length})` : ''}`} value='bailo' />
+              </Tabs>
+            </Box>
+            <Box data-test='modelListBox'>
+              {models &&
+                models.map((model: Model, index: number) => {
+                  const latestVersion = model.latestVersion as Version
+                  return (
+                    <Box key={model.uuid}>
+                      <Link href={`/model/${model.uuid}`} passHref legacyBehavior>
+                        <MuiLink
+                          variant='h5'
+                          sx={{ fontWeight: '500', textDecoration: 'none', color: theme.palette.secondary.main }}
+                        >
+                          {latestVersion.metadata.highLevelDetails.name}
+                        </MuiLink>
+                      </Link>
+                      <Typography variant='body1' sx={{ marginBottom: 2 }}>
+                        {latestVersion.metadata.highLevelDetails.modelInASentence}
+                      </Typography>
+                      <Stack direction='row' spacing={1} sx={{ marginBottom: 2 }}>
+                        {latestVersion.metadata.highLevelDetails.tags.map((tag: string) => (
+                          <Chip color='primary' key={`chip-${tag}`} label={tag} size='small' variant='outlined' />
+                        ))}
+                      </Stack>
+                      {index !== models.length - 1 && (
+                        <Box sx={{ borderBottom: 1, borderColor: 'divider', marginBottom: 2 }} />
+                      )}
+                    </Box>
+                  )
+                })}
+
+              {models?.length === 0 && <EmptyBlob data-test='emptyModelListBlob' text='No models here' />}
```

Comment:
Instead of doing `models?` here it would be better to fallback to an error message before getting to this point. That way we can treat `models` as a guaranteed array from that point onwards.

Reply:
I've added an error before but TS is still complaining unfortunately